zebra-chain/src/transaction.rs

```diff
@@ -395,26 +395,31 @@ impl Transaction {
 
     // orchard
 
-    /// Access the orchard::Nullifiers in this transaction, regardless of version.
-    pub fn orchard_nullifiers(&self) -> Box<dyn Iterator<Item = &orchard::Nullifier> + '_> {
-        // This function returns a boxed iterator because the different
-        // transaction variants can have different iterator types
+    /// Access the [`orchard::ShieldedData`] in this transaction, if the transaction has it,
+    /// regardless of version.
+    pub fn orchard_shielded_data(&self) -> Option<&orchard::ShieldedData> {
         match self {
-            // Actions
+            // Maybe Orchard shielded data
             Transaction::V5 {
-                orchard_shielded_data: Some(orchard_shielded_data),
+                orchard_shielded_data,
                 ..
-            } => Box::new(orchard_shielded_data.nullifiers()),
+            } => orchard_shielded_data.as_ref(),
 
-            // No Actions
+            // No Orchard shielded data
             Transaction::V1 { .. }
             | Transaction::V2 { .. }
             | Transaction::V3 { .. }
-            | Transaction::V4 { .. }
-            | Transaction::V5 {
-                orchard_shielded_data: None,
-                ..
-            } => Box::new(std::iter::empty()),
+            | Transaction::V4 { .. } => None,
+        }
+    }
+
+    /// Access the orchard::Nullifiers in this transaction, regardless of version.
+    pub fn orchard_nullifiers(&self) -> Box<dyn Iterator<Item = &orchard::Nullifier> + '_> {
+        // This function returns a boxed iterator because the different
+        // transaction variants can have different iterator types
+        match self.orchard_shielded_data() {
+            Some(orchard_shielded_data) => Box::new(orchard_shielded_data.nullifiers()),
+            None => Box::new(std::iter::empty()),
         }
     }
```

Comment:
`Option` is also an iterator:
```suggestion
    /// Access the orchard::Nullifiers in this transaction, regardless of version.
    pub fn orchard_nullifiers(&self) -> impl Iterator<Item = &orchard::Nullifier> {
        self.orchard_shielded_data().iter()
    }
```

Reply:
Isn't Rust fun?